### PR TITLE
Elide long context descriptions

### PR DESCRIPTION
**Pull request by fejta**
fixes https://github.com/kubernetes/test-infra/issues/10968
/assign @krzyzacy

### CODE REVIEW

File: prow/report/report.go [hunk 1]
```diff
@@ -78,8 +94,8 @@ func reportStatus(ghc GithubClient, pj v1.ProwJob, childDescription string) erro
 		}
 		if err := ghc.CreateStatus(refs.Org, refs.Repo, refs.Pulls[0].SHA, github.Status{
```
Comment (krzyzacy) [inline prow/report/report.go]: There are other calls to `ghc.CreateStatus`, do we want to move truncate inside of github client?

**Comment by krzyzacy** [comment]
truncate itself lgtm

File: prow/report/report.go [hunk 2]
```diff
@@ -66,6 +66,22 @@ func prowjobStateToGithubStatus(pjState v1.ProwJobState) (string, error) {
 	return "", fmt.Errorf("Unknown prowjob state: %v", pjState)
 }
 
+const (
+	maxLen = 140
```
Comment (krzyzacy) [inline prow/report/report.go]: consider add a comment/link to https://developer.github.com/v3/repos/deployments/#parameters-2

File: prow/report/report.go (hunk 1 above)
Comment (krzyzacy) [reply prow/report/report.go]: ok - we can wait until someone else also need to truncate :-)

File: prow/report/report.go (hunk 2 above)
Comment (fejta) [reply prow/report/report.go]: Done

File: prow/report/report.go [hunk 3]
```diff
@@ -66,6 +66,22 @@ func prowjobStateToGithubStatus(pjState v1.ProwJobState) (string, error) {
...
+const (
+	maxLen = 140 // https://developer.github.com/v3/repos/deployments/#parameters-2
+	elide  = " ... "
+)
+
+// truncate converts "really long messages" into "really ... messages".
+func truncate(in string) string {
+	const (
+		half = (maxLen - len(elide)) / 2
+	)
```
Comment (BenTheElder) [inline prow/report/report.go]: TIL you can use `const ( ... )` inside a func body! :O

**Pull request merged**